osu.Game/Beatmaps/Formats/LegacyDecoder.cs

```diff
@@ -203,9 +203,9 @@ public override HitSampleInfo ApplyTo(HitSampleInfo hitSampleInfo)
                 return baseInfo;
             }
 
-            public bool Equals(LegacySampleControlPoint other)
-                => base.Equals(other)
-                   && CustomSampleBank == other?.CustomSampleBank;
+            public override bool EquivalentTo(ControlPoint other) =>
+                base.EquivalentTo(other)
+                && CustomSampleBank == ((LegacySampleControlPoint)other).CustomSampleBank;
```

Comment:
Not going to pretend this isn't a nitpick, but is there any reason this is a straight cast, instead of a type-safe `is` check?

Reply:
Good catch, I thought the inheritance here was the other way around and that this was safe. It *is* currently safe due to `EquivalentTo` calls being very limited (and always type-checking first) but best not to do this.

Reply:
Why not making it protected? It has exact same signature with `Equals`.

Reply:
Worth `/// <inheritdoc />` on overrides, then the user won't get confused when calling on derived classes.

Reply:
equal checks the time as well